File: app/src/main/java/com/amaury/pointage/v2/engine/CelestialGlobeOrientationV2.kt
package com.amaury.pointage.v2.engine

/**
 * Display-only counter-rotation of the Earth around its own centre.
 * Input is the true-north heading already qualified by CelestialHeadingPolicyV2.
 * Never changes the GPS centre, geography, lighting, sky or clock hands.
 * Screen coordinates are +X right / +Y down on both mobile platforms.
 */
object CelestialGlobeOrientationV2 {
    fun counterRotationDeg(renderingHeadingDeg: Double?): Double {
        if (renderingHeadingDeg == null || !renderingHeadingDeg.isFinite()) return 0.0
        val heading = ((renderingHeadingDeg % 360.0) + 360.0) % 360.0
        // Equivalent angles close to north stay close to zero (359 -> +1).
        return if (heading <= 180.0) -heading else 360.0 - heading
    }
}

File: ios/HPTravail/HPTravail/CelestialV2/Core/CelestialGlobeOrientationV2.swift
import Foundation

/// Display-only counter-rotation around the Earth's own centre.
/// Input is true-north heading already qualified by CelestialHeadingPolicyV2.
/// GPS centre, geography, lighting, sky and clock hands are not changed.
/// Screen coordinates are +X right / +Y down on both mobile platforms.
enum CelestialGlobeOrientationV2 {
    static func counterRotationDegrees(renderingHeadingDegrees: Double?) -> Double {
        guard let value = renderingHeadingDegrees, value.isFinite else { return 0 }
        let remainder = value.truncatingRemainder(dividingBy: 360)
        let heading = (remainder + 360).truncatingRemainder(dividingBy: 360)
        // Equivalent angles close to north stay close to zero (359 -> +1).
        return heading <= 180 ? -heading : 360 - heading
    }
}
